### PR TITLE
TST: parametrize some indexing tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
2/many

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm

File: pandas/tests/indexing/test_floats.py [hunk 1]
```diff
@@ -59,115 +59,117 @@ def test_scalar_error(self, index_func):
...
             s.iloc[3.0] = 0
 
-    def test_scalar_non_numeric(self):
-
-        # GH 4892
-        # float_indexers should raise exceptions
-        # on appropriate Index types & accessors
-
-        for index in [
+    @pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: eventually consolidate move to top-level (as i suspect we have many of these fixtures, or will as you parametrize)

File: pandas/tests/indexing/test_floats.py [hunk 2]
```diff
@@ -59,115 +59,117 @@ def test_scalar_error(self, index_func):
...
-                        idxr(s)[3.0]
-
-                # label based can be a TypeError or KeyError
-                if s.index.inferred_type in {
-                    "categorical",
-                    "string",
-                    "unicode",
-                    "mixed",
-                }:
+            # getting
```
Comment (jreback) [inline pandas/tests/indexing/test_floats.py]: obviously this test can use ore parameterization, ok for later

File: pandas/tests/indexing/test_scalar.py [hunk 3]
```diff
@@ -9,61 +9,59 @@
 
 
 class TestScalar(Base):
-    def test_at_and_iat_get(self):
+    @pytest.mark.parametrize("kind", ["series", "frame"])
```
Comment (jreback) [inline pandas/tests/indexing/test_scalar.py]: make sure we use the same nomenclature, usually we call this klass and use the object itself

**Review by jreback — COMMENTED** [review]
ok for now, or can do more here, lmk.

**Comment by jbrockmendel** [comment]
> ok for now, or can do more here, lmk.

Agreed with all your comments.  These tests are a mess, (xref #31813, #31816) so id like to get these cleanups in small pieces to avoid merge headaches

**Comment by jreback** [comment]
thanks

**Pull request merged**